src/Spryker/Glue/AppPaymentBackendApi/Plugin/GlueApplication/PaymentConfirmDisconnectionRequestValidatorPlugin.php

```diff
@@ -0,0 +1,103 @@
+    {
+        return $this->getFactory()->getTranslatorFacade()->trans('Ignore & Disconnect');
+    }
+
+    protected function getLabelCancel(): string
+    {
+        return $this->getFactory()->getTranslatorFacade()->trans('Cancel');
+    }
+
+    protected function validateDisconnectionRequest(GlueRequestTransfer $glueRequestTransfer): GlueRequestValidationTransfer
+    {
+        if (empty($glueRequestTransfer->getMeta()[GlueRequestPaymentMapper::HEADER_TENANT_IDENTIFIER])) {
+            return (new GlueRequestValidationTransfer())
+                ->setIsValid(false)
+                ->addError(
+                    (new GlueErrorTransfer())
+                        ->setCode(AppPaymentBackendApiConfig::ERROR_CODE_PAYMENT_DISCONNECTION_TENANT_IDENTIFIER_MISSING)
+                        ->setMessage(
+                            $this->getFactory()->getTranslatorFacade()->trans('Tenant identifier is missing.'),
+                        ),
+                );
+        }
+
+        $paymentCollectionTransfer = $this->getFactory()->getAppPaymentFacade()->getPaymentCollection(
+            (new PaymentCriteriaTransfer())->setPaymentConditions(
+                (new PaymentConditionsTransfer())
+                    ->setTenantIdentifier($glueRequestTransfer->getMeta()[GlueRequestPaymentMapper::HEADER_TENANT_IDENTIFIER][0])
+                    ->setExcludingStatuses([
+                        PaymentStatus::STATUS_CANCELED,
+                        PaymentStatus::STATUS_SUCCEEDED,
+                    ]),
+            ),
+        );
+
+        if ($paymentCollectionTransfer->getPayments()->count() === 0) {
+            return (new GlueRequestValidationTransfer())
+                ->setIsValid(true);
+        }
+
+        return (new GlueRequestValidationTransfer())
+            ->setIsValid(false)
+            ->addError(
+                (new GlueErrorTransfer())
+                    ->setCode(AppPaymentBackendApiConfig::ERROR_CODE_PAYMENT_DISCONNECTION_CANNOT_BE_PROCEEDED)
+                    ->setMessage($this->getConfirmationErrorMessage()),
+            );
+    }
+
+    protected function onConfirmationOk(GlueRequestTransfer $glueRequestTransfer): GlueRequestValidationTransfer
+    {
+        return (new GlueRequestValidationTransfer())
+            ->setIsValid(true);
+    }
+
+    protected function onConfirmationCancel(GlueRequestTransfer $glueRequestTransfer): GlueRequestValidationTransfer
+    {
+        return (new GlueRequestValidationTransfer())
+            ->setIsValid(false)
+            ->setStatus(Response::HTTP_BAD_REQUEST)
+            ->addError(
+                (new GlueErrorTransfer())
+                    ->setCode(AppPaymentBackendApiConfig::ERROR_CODE_PAYMENT_DISCONNECTION_FORBIDDEN)
+                    ->setStatus(Response::HTTP_BAD_REQUEST)
+                    ->setMessage($this->getCancellationErrorMessage()),
+            );
+    }
+
+    protected function getConfirmationErrorMessage(): string
```

Comment:
From what I see I think that we could make it easier to implement by having default methods in the Abstract Plugin and the things changing as abstract methods. What changes looks like always the same the code and the message.

IMO it would be easier to have only

- `abstract function getCancellationErrorCode`
- `abstract function getCancellationErrorMessage`
etc

What do you think?

Reply:
OK, LGTM, fixed.